Pass --remote-layer-tree --use-gpu-process to run-api-tests on GPUP bots

heycam opened the pull request:
#### 98cdf6d7c6187f1f348803e8936724a95d1a0b9f
<pre>
Pass --remote-layer-tree --use-gpu-process to run-api-tests on GPUP bots
<a href="https://bugs.webkit.org/show_bug.cgi?id=250357">https://bugs.webkit.org/show_bug.cgi?id=250357</a>
rdar://103939125

Reviewed by Aakash Jain.

* Tools/CISupport/build-webkit-org/steps.py:
(RunAPITests.start):

Canonical link: <a href="https://commits.webkit.org/258800@main">https://commits.webkit.org/258800@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fcd554a972866ff149651561389d6d3a4a8722d2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102980 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12103 "Built successfully") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36003 "Failed to checkout and rebase branch from PR 8487") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/112231 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/172443 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106938 "Failed to checkout and rebase branch from PR 8487") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13125 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/3006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95217 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/110313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108753 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/10079 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/24814 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79436 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5536 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/26223 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5691 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/101866 "Passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/11699 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/45723 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7440 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3218 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -759,6 +759,11 @@ def start(self):
+        if additionalArguments:
+            for additionalArgument in additionalArguments:
+                if additionalArgument in ["--remote-layer-tree", "--use-gpu-process"]:
+                    self.command += additionalArgument
```
This results in each character being added individually to self.command list. e.g.: https://build.webkit-dev.org/#/builders/151/builds/3/steps/19/logs/stdio

changing `additionalArgument` to `[additionalArgument]`  works fine

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -759,6 +759,11 @@ def start(self):
+        additionalArguments = self.getProperty("additionalArguments")
+        if additionalArguments:
+            for additionalArgument in additionalArguments:
+                if additionalArgument in ["--remote-layer-tree", "--use-gpu-process"]:
```
Nit: might be a good idea to define these arguments in the beginning of the class, e.g.: `VALID_ADDITIONAL_ARGUMENTS_LIST = ["--remote-layer-tree", "--use-gpu-process"]`

aj062 commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -759,6 +759,11 @@ def start(self):
         self.failedTestCount = 0
         appendCustomTestingFlags(self, self.getProperty('platform'), self.getProperty('device_model'))
+        additionalArguments = self.getProperty("additionalArguments")
+        if additionalArguments:
```
Nit: can skip this if condition by changing `self.getProperty("additionalArguments")` to `self.getProperty("additionalArguments", [])`

aj062 reviewed (commented):
Test run with this change on dev instance: https://build.webkit-dev.org/#/builders/151/builds/5 (vpn might be required to access the above url)

heycam committed:
Pass --remote-layer-tree --use-gpu-process to run-api-tests on GPUP bots
https://bugs.webkit.org/show_bug.cgi?id=250357
rdar://103939125

Reviewed by Aakash Jain.

* Tools/CISupport/build-webkit-org/steps.py:
(RunAPITests.start):

Canonical link: https://commits.webkit.org/258800@main
